### PR TITLE
CLOUDP-199915: Use go-driver search index management helpers instead of createSearchIndex command

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes

Use go-driver search index management helpers instead of createSearchIndexCommand

CLOUDP-199915

### CODE REVIEW

**Comment by jeroenvervaeke** [comment]
This PR is ready when https://github.com/mongodb/mongo-go-driver/pull/1647 is merged.

Before merging, remove the override to my personal repo.

**Comment by jeroenvervaeke** [comment]
`mongo-go-driver` was released today, which includes the functionality to specify the search index type.
https://github.com/mongodb/mongo-go-driver/releases/tag/v1.16.0

**Review by gssbzn — APPROVED** [review]
LGTM if LGTEVG

**Pull request merged**
